app/models/restore_transfer.rb

```diff
@@ -19,9 +19,9 @@ def self.find_fields
     Set.new [:restore_id]
   end
 
-  def cancel!(reason)
+  def cancel!(reason, detail)
     unless cancelled
-      update!(cancelled: true, cancel_reason: reason)
+      update!(cancelled: true, cancel_reason: reason, cancel_reason_detail: detail)
     end
   end
```

Comment:
The simplicity is a lie! -> The automation for RestoreTransfers isn't implemented.